File: calculator.lua
-- Function to perform calculations
function calculate(num1, num2, operation)
    if operation == "+" then
        return num1 + num2
    elseif operation == "-" then
        return num1 - num2
    elseif operation == "*" then
        return num1 * num2
    elseif operation == "/" then
        if num2 ~= 0 then
            return num1 / num2
        else
            return "Error: Division by zero"
        end
    else
        return "Error: Unsupported operation"
    end
end

-- Main program loop
while true do
    print("Simple Calculator")
    
    io.write("Enter first number: ")
    local num1 = tonumber(io.read())
    
    io.write("Enter operation (+, -, *, /): ")
    local operation = io.read()
    
    io.write("Enter second number: ")
    local num2 = tonumber(io.read())
    
    local result = calculate(num1, num2, operation)
    print("Result: " .. tostring(result))
    
    print("Press Enter to continue or type 'exit' to quit.")
    local input = io.read()
    if input == "exit" then
        break
    end
end

File: calculator.cpp
#include <iostream>
#include <stdexcept>

using namespace std;

// Function to perform calculations
double calculate(double num1, double num2, char operation) {
    switch (operation) {
        case '+':
            return num1 + num2;
        case '-':
            return num1 - num2;
        case '*':
            return num1 * num2;
        case '/':
            if (num2 != 0) {
                return num1 / num2;
            } else {
                throw runtime_error("Error: Division by zero");
            }
        default:
            throw invalid_argument("Error: Unsupported operation");
    }
}

int main() {
    cout << "Simple Calculator" << endl;

    while (true) {
        try {
            double num1, num2;
            char operation;

            cout << "Enter first number: ";
            cin >> num1;

            cout << "Enter operation (+, -, *, /): ";
            cin >> operation;

            cout << "Enter second number: ";
            cin >> num2;

            double result = calculate(num1, num2, operation);
            cout << "Result: " << result << endl;

            cout << "Do you want to perform another calculation? (yes/no): ";
            string continueCalc;
            cin >> continueCalc;
            if (continueCalc != "yes") {
                break;
            }
        } catch (const runtime_error& e) {
            cout << e.what() << endl;
        } catch (const invalid_argument& e) {
            cout << e.what() << endl;
        } catch (...) {
            cout << "An unexpected error occurred." << endl;
        }
    }

    cout << "Goodbye!" << endl;
    return 0;
}
